backends-velox/src/main/scala/org/apache/gluten/execution/HashAggregateExecTransformer.scala

```diff
@@ -60,6 +60,12 @@ abstract class HashAggregateExecTransformer(
     resultExpressions,
     child) {
 
+  override def output: Seq[Attribute] = {
+    // TODO: We should have a check to make sure the returned schema actually matches the output
+    //  data. Since "resultExpressions" is not actually in used by Velox.
+    super.output
```

Comment:
Why do we need to highlight this, did not we add post project with `resultExpressions` ?

Reply:
The `projectList` of agg post project is the `resultExpressions`.

Reply:
> The `projectList` of agg post-project is the `resultExpressions`. Is the intended expression here something like, when post-project hasn't been inserted, the output of `HashAggregateExecTransformer` isn't actually equal to `resultExpressions`?

That's correct. We don't have strict verification between `groupingExpressions+aggregateExpressions` and `resultExpressions` to ensure them matching together when instantiating `HashAggregateExecTransformer`, so there will be risk that unexpected errors happens at runtime when a `HashAggregateExecTransformer` is created without pre/post project. 

I think it will be fine if we can have a check for `needsPreProjection == false & needsPostProjection == false` at planning time in `HeuristicApplier` as well to fix this (I used very similar code to pick executable plan in RAS in this PR) in future.

Anyway it's just a soft `TODO` since current code doesn't actually impact anything.